app/services/trainees/create_from_apply.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+module Trainees
+  class CreateFromApply
+    include ServicePattern
+
+    def initialize(json: example_json)
+      @attributes = json["attributes"]
+    end
+
+    def call
+      trainee.save!
+      trainee
+    end
+
+  private
+
+    attr_reader :attributes
+
+    def example_json
+      JSON.parse(example_file)
+    end
+
+    def example_file
+      File.read(Rails.root.join("lib/example_apply_response.json"))
+    end
+
+    def trainee
+      @trainee ||= Trainee.new(mapped_attributes)
+    end
+
+    def mapped_attributes
+      {
+        # TODO: Use actual provider, if found.
+        provider: Provider.find(1),
+        trainee_id: raw_trainee["id"],
```

Comment:
a bit annoying, but it is what it is...

Reply:
What's the matter here @defong ? Can we improve on the Apply side?

Reply:
`trainee_id` nothing to do with Apply, we just have an identifier or field that is solely of questionable nature. 

Unless on Apply, the provider additional gives the candidate another symbolic name, such as `cohort 2021`, to remove the need to refer it by PII.

Reply:
We might want to rename our field anyway to make it clear it's the provider's own internal id for the trainee.